### PR TITLE
Change 'search for a prisoner' function in manage key workers to elastic search

### PR DESCRIPTION
**Pull request by steverendell**
Replaced offender search, sentence and csra data prison-api api calls with prisoner-search calls (Elastic search)

### CODE REVIEW

File: backend/api/oauthEnabledClient.js [hunk 1]
```diff
@@ -16,7 +16,8 @@ const errorLogger = (error) => {
   const responseData = error.response ? error.response.body : '-'
   if (error.response && error.response.req) {
     logger.warn(
-      `API error in ${error.response.req.method} ${error.response.req.path} ${status} ${error.message} ${responseData}`
+      { message: error.message, responseData },
+      `API error in ${error.response.req.method} ${error.response.req.path} ${status}`
```
Comment (steverendell) [inline backend/api/oauthEnabledClient.js:20]: error.message and responseData are not strings

File: backend/services/allocationService.js [hunk 2]
```diff
@@ -80,26 +72,33 @@ const serviceFactory = (elite2Api, keyworkerApi, offenderSearchResultMax) => {
         ...offenderWithAllocatedKeyworker,
         keyworkerDisplay,
         numberAllocated,
-        crsaClassification: findCrsaForOffender(allCsraRatings, offenderNo),
-        confirmedReleaseDate: findReleaseDateForOffender(allReleaseDates, offenderNo),
+        ...findForOffender(allPrisoners, offenderNo),
```
Comment (steverendell) [inline backend/services/allocationService.js:75]: release date and csra info are already in the data from prisoner-search

File: backend/services/allocationService.js [hunk 3]
```diff
@@ -256,14 +252,18 @@ const serviceFactory = (elite2Api, keyworkerApi, offenderSearchResultMax) => {
     }
   }
 
-  const searchOffendersPaginated = async (context, { agencyId, keywords, locationPrefix, pageRequest }) => {
-    const availableKeyworkers = await keyworkerApi.availableKeyworkers(context, agencyId)
+  const searchOffendersPaginated = async (context, { agencyId, locationPrefix, pageRequest }) => {
+    const systemContext = await systemOauthClient.getClientCredentialsTokens(context.username)
```
Comment (steverendell) [inline backend/services/allocationService.js:256]: Sharing API_SYSTEM_CLIENT_ID with the complex needs api

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**